revenger/infrastructure/python_adapter_v2.py

```diff
@@ -0,0 +1,212 @@
+                self.logger.log_debug(f'  Created sub type {member_sub_type} from filemodule: >{filemodule}<')
+                self.logger.log_debug(
+                    f'  Type {member_sub_type} not found in {type_dict.keys()} saving as type from module {member_sub_type}')
+            return member_sub_type
+
+        # TODO: Move it to classDef itself and improve implementation
+        def analyse_class_def(self, node: ClassDef) -> Tuple[GenericSubDataStructure | None, str]:
+            active_class_name: str = None
+
+            if len(self.active_class_name_list) > 0:
+                active_class_name = self.active_class_name_list[-1]
+
+            fqdn_class_name: str = f'{self.filemodule}.{node.name}'
+            if active_class_name is not None:
+                fqdn_class_name = f'{self.filemodule}.{active_class_name}.{node.name}'
+
+            parent_sub_datastructure: GenericSubDataStructure = self.datastructure.get_datastructures_from_class_name(
+                self.get_current_class_name())
+
+            if parent_sub_datastructure is not None:
+                parent_sub_datastructure.add_inner_class(fqdn_class_name)
+
+            self.datastructure.append_class(self.filename, self.filemodule, self.from_import, fqdn_class_name, self.namespace_list)
+
+            sub_datastructure: GenericSubDataStructure = self.datastructure.get_datastructures_from_class_name(fqdn_class_name)
+            if sub_datastructure is not None:
+                for base in node.bases:
+                    if isinstance(base, ast.Name):
+                        base_class = base.id
+                        if base_class == 'ABC':
+                            sub_datastructure.set_abstract()
+                        else:
+                            sub_datastructure.add_base_class(base_class)
+
+            return sub_datastructure, fqdn_class_name
+
+
+        def visit_ImportFrom(self, node: ImportFrom) -> Any:
+            module_path = node.module
+
+            for module_name in node.names:
+                self.from_import[module_name.name] = module_path + '.' + module_name.name
+
+            self.logger.log_warn(f' {self.from_import}')
+            ast.NodeVisitor.generic_visit(self, node)
+
+        def generic_visit(self, node):
+            # print(node)
+            ast.NodeVisitor.generic_visit(self, node)
+
+        def visit_ClassDef(self, node: ClassDef) -> Any:
+
+            sub_datastructure, fqdn_class_name = self.analyse_class_def(node)
+
+            self.active_class_name_list.append(fqdn_class_name)
+            ast.NodeVisitor.generic_visit(self, node)
+
+            if len(self.active_class_name_list) > 0:
+                self.active_class_name_list.pop()
+
+        def visit_AnnAssign(self, node: AnnAssign) -> Any:
+            static_name: str = node.target.id
+            static_type: str = CommonInfrastructure.NOT_EXTRACTED
+            if isinstance(node.annotation, ast.Name):
```

Comment:
I was wondering whether moving check of name in a specific leave of the tree would make sense. What do you think?